adr-002.md

```diff
@@ -0,0 +1,171 @@
+consulted:
+informed:
+---
+
+# ADR 002: Directory Structure for System Contracts Repository
+
+## Context and problem statement
+
+The new Hedera System Contracts repository needs a consistent, readable, and scalable directory structure. The structure should make it easy for developers to find production contracts, interfaces, and their extensions while allowing for future growth.
+
+We need to decide between a simplified flat structure versus a more granular modular split.
+
+## Decision Drivers
+
+- Ease of finding production contracts and interfaces
+- Consistency with community practices (e.g., OpenZeppelin Contracts)
+
+## Considered Options
+
+1. **Simplified Structure** (recommended)
+2. Modular, Granular Structure
+
+## Option 1: Simplified Structure
+
+A flatter structure with fewer nested directories, making it easier to navigate and understand at a glance.
+
+**Directory Layout:**
+
+```
+contracts/
+├── account-service/
+│   ├── HederaAccountService.sol
+│   ├── IHederaAccountService.sol
+│   ├── IHRC632.sol
+│   ├── IHRC904AccountFacade.sol
+│   ├── IHRC906.sol
+│   └── IHRC906AccountFacade.sol
+├── schedule-service/
+│   ├── HederaScheduleService.sol
+│   ├── IHederaScheduleService.sol
+│   ├── IHRC755.sol
+│   ├── IHRC755ScheduleFacade.sol
+│   └── IHRC756.sol
+├── token-service/
+│   ├── HederaTokenService.sol
+│   ├── IHederaTokenService.sol
+│   ├── IHRC719.sol
+│   ├── IHRC904.sol
+│   ├── IHRC904TokenFacade.sol
+│   ├── IHTSStructs.sol
+│   ├── FeeHelper.sol
+│   ├── KeyHelper.sol
+│   ├── ExpiryHelper.sol
+│   └── safe-hts/
+├── token-service-v2/
+│   ├── HederaTokenService.sol
+│   ├── IHederaTokenServiceV2.sol
+│   ├── FeeHelper.sol
+│   ├── KeyHelper.sol
+│   └── HederaResponseCodes.sol
+├── prng/
+│   ├── PrngSystemContract.sol
+│   └── IPrngSystemContract.sol
+├── common/
+│   └── HederaResponseCodes.sol
+└── mocks/
+test/
+```
+What did we remove:
+- contracts/system-contracts/hedera-account-service/examples
+- contracts/system-contracts/hedera-schedule-service/examples
+- contracts/system-contracts/hedera-token-service/examples
+- contracts/system-contracts/hedera-token-service/AtomicHTS.sol
+- contracts/system-contracts/hedera-token-service-v2/examples
+- contracts/system-contracts/native/EcrecoverCaller.sol
+- contracts/system-contracts/native/EthNativePrecompileCaller.sol
+Why: these usage examples would fit better in another (sample/tools) repository.
+
+## Consequences
+
+### Positive
+
+- Simple, flat structure - easy to navigate
+
+### Negative
+
+- Less separation between different types of files (interfaces, helpers, implementations)
+
+## Option 2: Modular, granular structure
+
+A more hierarchical structure with dedicated subdirectories for interfaces, utilities, and other components.
+
+**Directory Layout:**
+
+```
+contracts/
+├── interfaces/              # Top-level facades
```

Comment:
I dont see the interface split in one folder.  I think we need to wrap up for comprehenion of the full context